src/service/controllers/conf.js

```diff
@@ -370,6 +371,24 @@ copyActiveSessionToAvailability = (agent, a) => {
   }
 };
 
+setMonitoringState = (req, a) => {
+  if (a.isMonitored) {
+    return;
+  }
+  const user = exports.userByCode(req);
+  const userDn = ((user || {}).activeSession || {}).dn || {};
+  if (userDn.agentState !== 'Ready' || _.indexOf(userDn.capabilities, 'start-monitoring') === -1) {
+    return;
+  }
+  const monitoredAG = _.find(agentGroups, ag => {
+    return _.indexOf(ag.managerDBIDs, user.DBID) !== -1;
+  });
+  if (monitoredAG && _.indexOf(monitoredAG.agentDBIDs, a.DBID) !== -1) {
+    const channels = a.availability ? a.availability.channels : [];
+    a.isMonitorable = !_.isUndefined(_.find(channels, ch => { return ch.name === 'voice' && userDn.switchName === ch.switchName; }));
```

Comment:
If you use _.some here instead of _.find, you don't need to use _.isUndefined, as _.some returns a boolean